src/codex_query.py

```diff
@@ -140,68 +127,62 @@ def get_query(prompt_file):
 
     # get input from terminal or stdin
     if DEBUG_MODE:
-        entry = input("prompt: ") + '\n'
+        entry = input("prompt: ")
     else:
-        entry = sys.stdin.read()
+        # Remove extreaneous newlines and hashtag from the input string
+        entry = sys.stdin.read().strip("\n").strip("#").strip(" ") 
     # first we check if the input is a command
-    command_result, prompt_file = get_command_result(entry, prompt_file)
+    try:
+        command_result, prompt_generator = get_command_result(entry, prompt_generator)
+    except Exception as e:
+        print (str(e))
 
     # if input is not a command, then query Codex, otherwise exit command has been run successfully
     if command_result == "":
-        return entry, prompt_file
+        return entry, prompt_generator
     else:
         sys.exit(0)
 
-def detect_shell():
-    global SHELL
-    global PROMPT_CONTEXT
 
+def get_shell_prefix(shell: str):
+    # prime codex for the corresponding shell type
+    if shell == "zsh":
+        return '#!/bin/zsh\n\n'
+    elif shell == "bash":
+        return '#!/bin/bash\n\n'
```

Comment:
I don't think this is the most appropriate way (same for zsh). I think the code should use `#!/usr/bin/env bash`

Reply:
The above if condition is the same as the one in the previous implementation of codex-cli, I can change that, but we'll have to run tests with the updated prefix